### PR TITLE
[Merged by Bors] - chore: use #adaptation_note for labelling adaptation notes

### PR DESCRIPTION
**Pull request by grunweg**
---

- [x] depends on: #12662

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Free.lean [hunk 1]
```diff
@@ -85,11 +86,11 @@ theorem hom_ext {β : Type v} [Mul β] {f g : FreeMagma α →ₙ* β} (h : f 
 
 end FreeMagma
 
+#adaptation_note /-- around nightly-2024-02-25, we need to write `mul x y` in the second pattern,
+instead of `x * y`. --/
 /-- Lifts a function `α → β` to a magma homomorphism `FreeMagma α → β` given a magma `β`. -/
 def FreeMagma.liftAux {α : Type u} {β : Type v} [Mul β] (f : α → β) : FreeMagma α → β
   | FreeMagma.of x => f x
-  -- Adaptation note: around nightly-2024-02-25, we need to write `mul x y` in the pattern here,
```
Comment (grunweg) [inline Mathlib/Algebra/Free.lean:91]: Writing #adapation_note here errors, hence I moved this comment before the function.

File: Mathlib/Algebra/Homology/Homotopy.lean [hunk 2]
```diff
@@ -542,9 +542,8 @@ def mkInductiveAux₂ :
     ⟨(P.xNextIso rfl).hom ≫ I.1, I.2.1 ≫ (Q.xPrevIso rfl).inv, by simpa using I.2.2⟩
 #align homotopy.mk_inductive_aux₂ Homotopy.mkInductiveAux₂
 
--- Adaptation note:
--- During the port we marked these lemmas with `@[eqns]` to emulate the old Lean 3 behaviour.
--- See https://github.com/leanprover-community/mathlib4/issues/11647
+-- Porting note(#11647): during the port we marked these lemmas
```
Comment (grunweg) [inline Mathlib/Algebra/Homology/Homotopy.lean:545]: I think these should rather be porting notes. (Also applies in a few more places.)
I could be wrong about these, though!

File: Mathlib/Analysis/Complex/UpperHalfPlane/Basic.lean [hunk 3]
```diff
@@ -8,6 +8,7 @@ import Mathlib.LinearAlgebra.Matrix.SpecialLinearGroup
 import Mathlib.Analysis.Complex.Basic
 import Mathlib.GroupTheory.GroupAction.Defs
 import Mathlib.LinearAlgebra.Matrix.GeneralLinearGroup
+import Mathlib.Tactic.AdaptationNote
```
Comment (grunweg) [inline Mathlib/Analysis/Complex/UpperHalfPlane/Basic.lean:11]: General question: should AdaptationNote be added to Tactic.Common? (I'm fine with not. The current situation added a little bit of fraction to the conversion, but that was a one-time event.)

Comment (semorrison) [reply Mathlib/Analysis/Complex/UpperHalfPlane/Basic.lean:11]: Let's leave it out for now.

**Comment by semorrison** [comment]
Thanks! Please wait for the upstream PR before merging. 

bors d+

**Comment by grunweg** [comment]
This currently fails the "find #-commands" step.
@adomani As the author of #10809, do you happen to have some spare time to help?
Do you know what needs to be done so `#help` also mentions #adaptation_note, or that one is allowed? Thank you!

**Comment by grunweg** [comment]
(Of course, when #11019 lands, such an exception can also be made in the Lean version.)

**Comment by adomani** [comment]
Michael, I did not realize that you posted this here as well.  #12675 allows adaptation notes.  I'm not sure that I understand your question about `#help`: it should pick up adaptation notes on its own, I think.

**Comment by grunweg** [comment]
Sorry for the double-post; I realised later that #12662 exposed this already.
Given there's a fix for the linter, whatever question I had is resolved: essentially I was asking how to fix this.

**Pull request closed without merging**